geolocator_platform_interface/lib/src/models/location_settings.dart

```diff
@@ -0,0 +1,54 @@
+import '../enums/location_accuracy.dart';
+
+/// Represents the abstract [LocationSettings] class with which you can
+/// configure platform specific settings.
+class LocationSettings {
+  /// Initializes a new [LocationSettings] instance with default values.
+  ///
+  /// The following default values are used:
+  /// - accuracy: best
+  /// - distanceFilter: 0
+  /// - forceAndroidLocationManager: false
+  /// - timeInterval: 0
+  /// - timeLimit: 0
+  const LocationSettings(
+      {this.accuracy = LocationAccuracy.best,
+      this.distanceFilter = 0,
+      this.timeInterval = 0,
+      this.timeLimit});
+
+  /// Defines the desired accuracy that should be used to determine the
+  /// location data.
+  ///
+  /// The default value for this field is [LocationAccuracy.best].
+  final LocationAccuracy accuracy;
+
+  /// The minimum distance (measured in meters) a device must move
+  /// horizontally before an update event is generated.
+  ///
+  /// Supply 0 when you want to be notified of all movements. The default is 0.
+  final int distanceFilter;
+
+  /// The desired interval for active location updates, in milliseconds
+  /// (Android only).
+  ///
+  /// On iOS this value is ignored since position updates based on time
+  /// intervals are not supported.
+  final int timeInterval;
```

Comment:
Since we are making a breaking change, it would be better to change this to a `Duration` type (similar to `timeLimit` parameter). And also rename this parameter to `intervalDuration`:
```suggestion
  final Duration? intervalDuration;
```